src/ucp/wireup/address.c

```diff
@@ -1653,6 +1637,24 @@ static ucp_rsc_index_t ucp_address_get_remote_device_index(
     return ucs_array_length(device_array) - 1;
 }
 
+static void
+ucp_address_unpacked_adjust_legacy(ucp_unpacked_address_t *unpacked_address)
+{
+    uint8_t frozen_minor_api_ver = ucp_address_packed_release_version(18);
+    size_t i;
+
+    if ((unpacked_address->addr_version != UCP_OBJECT_VERSION_V1) ||
+        (unpacked_address->dst_version > frozen_minor_api_ver)) {
```

Comment:
why do you compare unpacked version with packed version?

Reply:
the unpacked version is a version of the peer, if it's old one, then reset legacy bits to calculate correct md_index with extended mask

Reply:
i mean, that for the peer you take **unpacked** format and compare it with **packed** current version

Reply:
packed current (18) is the last version that contains legacy bits, after that no need to reset them

Reply:
can you implement some gtest to check?
AFAIU, if peer's adress version is 1.17, unpacked peer version will be 17, but packed current will be 2. In this case md flag bits will not be cleared in md index which is wroing.

Reply:
I see your point, need to compare to `ucp_address_unpack_release_version(frozen_minor_api_ver)`